### PR TITLE
Docs: Setup OAuth providers using the SSO Settings UI 

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by sarahleejane — COMMENTED** [review]
If I am a cloud customer, all of my options are in private preview only.  
Perhaps worth adding an extra option for now, to go to gcom advanced auth and manage from there?

File: docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md [hunk 1]
```diff
@@ -18,12 +18,61 @@ weight: 1400
...
+## Configure Okta authentication client using the Grafana UI
 
-## Steps
+{{% admonition type="note" %}}
+Available in Public Preview in Grafana 10.4 behind the `ssoSettingsApi` feature toggle.
+{{% /admonition %}}
+
+As a Grafana Admin, you can configure generic OAuth2 client from within Grafana using the Okta UI. To do this, navigate to **Administration > Authentication > Okta** page and fill in the form. If you have a current configuration in the Grafana configuration file then the form will be pre-populated with those values otherwise the form will contain default values.
+
+After you have filled in the form, click **Save** to save the configuration. If the save was successful, Grafana will apply the new configurations.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md:31]:
```suggestion
After you have filled in the form, click **Save**. If the save was successful, Grafana will apply the new configurations.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md [hunk 2]
```diff
@@ -18,12 +18,61 @@ weight: 1400
...
+
+In case you would like to reset your changes you made on the UI back to the default values, click **Reset**. After you have reset the changes, Grafana will apply the configuration from the Grafana configuration file (if there is any configuration) or the default values.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md:33]:
```suggestion
If you need to reset changes you made in the UI back to the default values, click **Reset**. After you have reset the changes, Grafana will apply the configuration from the Grafana configuration file (if there is any configuration) or the default values.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md [hunk 3]
```diff
@@ -31,7 +31,56 @@ First, you need to create a Google OAuth Client:
...
+    hosted_domain   = "mycompany.com"
+    use_pkce        = true
+  }
+}
+```
+
+Go to [Terraform Registry](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings) for a complete reference on using the `grafana_sso_settings` resource.
+
+## Configure Google authentication client using the Grafana configuration file
+
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:78]: Please add a sentence or 2, we do not want headings without anything underneath.

File: docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md [hunk 4]
```diff
@@ -25,12 +25,59 @@ This topic describes how to configure GitHub OAuth2 authentication.
...
-## Steps
+{{% admonition type="note" %}}
+Available in Public Preview in Grafana 10.4 behind the `ssoSettingsApi` feature toggle.
+{{% /admonition %}}
+
+As a Grafana Admin, you can configure generic OAuth2 client from within Grafana using the GitHub UI. To do this, navigate to **Administration > Authentication > GitHub** page and fill in the form. If you have a current configuration in the Grafana configuration file then the form will be pre-populated with those values otherwise the form will contain default values.
+
+After you have filled in the form, click **Save** to save the configuration. If the save was successful, Grafana will apply the new configurations.
+
+In case you would like to reset your changes you made on the UI back to the default values, click **Reset**. After you have reset the changes, Grafana will apply the configuration from the Grafana configuration file (if there is any configuration) or the default values.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md:40]:
```suggestion
If you need to reset changes you made in the UI back to the default values, click **Reset**. After you have reset the changes, Grafana will apply the configuration from the Grafana configuration file (if there is any configuration) or the default values.
```

**Review by lwandz13 — CHANGES_REQUESTED** [review]

**Review by Jguer — APPROVED** [review]

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 5]
```diff
@@ -25,12 +25,61 @@ This topic describes how to configure GitLab OAuth2 authentication.
...
+    use_pkce              = true
+    use_refresh_token     = true
+  }
+}
+```
+
+Go to [Terraform Registry](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings) for a complete reference on using the `grafana_sso_settings` resource.
+
+## Configure GitLab authentication client using the Grafana configuration file
+
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:77]: Changed the structure a little bit (removed the `Before you begin` subtitle).

File: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md (hunk 3 above)
Comment (mgyongyosi) [reply docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:78]: Changed the structure a little bit (removed the `Before you begin` subtitle).

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 6]
```diff
@@ -161,7 +161,60 @@ If the setting is set to `false`, the user is assigned the role of `Admin` of th
...
+}
+```
+
+Refer to [Terraform Registry](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/sso_settings) for a complete reference on using the `grafana_sso_settings` resource.
+
+## Configure Azure AD/Entra ID authentication client using the Grafana configuration file
+
+Ensure that you have access to the [Grafana configuration file]({{< relref "../../../configure-grafana#configuration-file-location" >}}).
+
+### Enable Azure AD/Entra ID OAuth in Grafana
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:217]:
```suggestion
### Enable Microsoft Entra ID OAuth in Grafana
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 7]
```diff
@@ -13,15 +13,15 @@ labels:
     - enterprise
     - oss
 menuTitle: Azure AD OAuth2
-title: Configure Azure AD OAuth2 authentication
+title: Configure Azure AD/Entra ID OAuth2 authentication
 weight: 800
 ---
 
-# Configure Azure AD OAuth2 authentication
+# Configure Azure AD/Entra ID OAuth2 authentication
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:20]:
```suggestion
# Configure Microsoft Entra ID OAuth2 authentication
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 8]
```diff
@@ -13,15 +13,15 @@ labels:
...
 
 The Azure AD authentication allows you to use an Azure Active Directory tenant as an identity provider for Grafana. You can use Azure AD application roles to assign users and groups to Grafana roles from the Azure Portal.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:22]: I'd call out here that Azure AD is not called Microsoft Entra ID

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 9]
```diff
@@ -13,15 +13,15 @@ labels:
...
 
-## Create the Azure AD application
+## Create the Azure AD/Entra ID application
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:24]:
```suggestion
## Create the Microsoft Entra ID application
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 10]
```diff
@@ -161,7 +161,60 @@ If the setting is set to `false`, the user is assigned the role of `Admin` of th
 }
 ```
 
-## Enable Azure AD OAuth in Grafana
+## Before you begin
+
+Ensure that you have followed the steps in [Create the Azure AD/Entra ID application](#create-the-azure-adentra-id-application) before you begin.
+
+## Configure Azure AD/Entra ID authentication client using the Grafana UI
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:168]:
```suggestion
## Configure Microsoft Entra ID authentication client using the Grafana UI
```

**Review by lwandz13 — CHANGES_REQUESTED** [review]

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md (hunk 8 above)
Comment (mgyongyosi) [reply docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md]: https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id says that Azure AD is Microsoft Entra ID, but we'll update these in a follow up PR.

**Review by lwandz13 — APPROVED** [review]

**Pull request merged**